dali/daliadmin/daliadmin.cpp

```diff
@@ -171,6 +171,9 @@ static const char *splitpath(const char *path,StringBuffer &head,StringBuffer &t
 {
     if (path[0]!='/')
         path = tmp.append('/').append(path).str();
+    int len = strlen(path);
+    if (len > 1 && path[len-1] == '/')
+        WARNLOG("xpath here shouldn't end with '/'");
```

Comment:
Since all use cases require a tail, it would be better to fire an exception with the path and let daliadmin's catch trace the error.
Then you can also remove the if (!tail) return; calls in all the functions.
Something like:
```
    const char *tail = splitXPath(path, head);
    if (isEmptyString(tail))
        throw MakeStringException(0, "Expecting xpath tail node in: %s", path);
    return tail;
```